agent/sshd/cmd_docker.go

```diff
@@ -30,21 +30,54 @@ func newCmd(u *osauth.User, shell, term, host string, command ...string) *exec.C
 func nsenterCommandWrapper(uid, gid uint32, home string, command ...string) ([]string, error) {
 	wrappedCommand := []string{}
 
+	const baseNsenterPath string = "/proc/1/ns/"
+
+	nsenterPaths := map[string]string{
+		"cgroup": "-C",
+		"ipc":    "-i",
+		"mnt":    "-m",
+		"pid":    "-p",
+		"time":   "-T",
+		"user":   "-U",
+		"uts":    "-u",
+	}
+
+	nsenterArgs := []string{}
+
+	for path, params := range nsenterPaths {
+		if _, err := os.Stat(baseNsenterPath + path); err != nil {
+			continue
+		}
+		nsenterArgs = append(nsenterArgs, []string{params, baseNsenterPath + path}...)
+	}
+
 	if _, err := os.Stat("/usr/bin/nsenter"); err == nil {
-		wrappedCommand = append([]string{
-			"/usr/bin/setpriv",
-			"--init-groups",
-			"--ruid",
-			strconv.Itoa(int(uid)),
-			"--regid",
-			strconv.Itoa(int(gid)),
-			"/usr/bin/nsenter",
-			"-t", "1",
-			"-a",
-			"-S",
-			strconv.Itoa(int(uid)),
-			fmt.Sprintf("--wd=%s", home),
-		}, wrappedCommand...)
+		wrappedCommand = func(wc []string, nsArgs []string) []string {
+			setPrivCommands := []string{
+				"/usr/bin/setpriv",
+				"--init-groups",
+				"--ruid",
+				strconv.Itoa(int(uid)),
+				"--regid",
+				strconv.Itoa(int(gid)),
+			}
+
+			nssenterCommands := append([]string{
+				"/usr/bin/nsenter",
+				"-t",
+				"1",
+			}, nsArgs...)
+
+			nssenterCommands = append(nssenterCommands,
+				[]string{
+					"-S",
+					strconv.Itoa(int(uid)),
+					fmt.Sprintf("--wd=%s", home),
+				}...,
+			)
+
+			return append(setPrivCommands, nssenterCommands...)
+		}(wrappedCommand, nsenterArgs)
```

Comment:
I'd move this to a private function on this module, so it avoids this big anonymous function, confusing to understand.